libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt

```diff
@@ -37,7 +37,11 @@ class MapboxNavigationCameraTransition(
         transitionOptions: NavigationCameraTransitionOptions
     ): AnimatorSet {
         val pluginImpl: CameraAnimationsPluginImpl? = cameraPlugin as? CameraAnimationsPluginImpl
+        val isTransitionImmediate = transitionOptions.maxDuration == 0L
 
+        if (isTransitionImmediate) {
+            return fromLowZoomToHighZoom(cameraOptions, transitionOptions)
+        }
```

Comment:
Why do we need to fallback to the old impl? Why does `constraintDurationTo(transitionOptions.maxDuration)` not work?
https://github.com/mapbox/mapbox-navigation-android/blob/74f1ccf1a172c7b067942fdba3b5ac14df9c8c4d/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt#L317-L319

The extension should constraint the duration to `0` and the transition should be instant.

Reply:
This does not work because `constraintDurationTo` applies the constraint to child animators within the animator set and not the animator set itself. 
```
childAnimations.maxByOrNull { it.startDelay + it.duration }?.let {
        val longestExecutionTime = it.startDelay + it.duration
        if (longestExecutionTime > maxDuration) {
            val factor = maxDuration / (longestExecutionTime).toDouble()
            childAnimations.forEach { animator ->
                animator.startDelay = (animator.startDelay * factor).toLong()
                animator.duration = (animator.duration * factor).toLong()
            }
        }
    }
```
Since we call `setDuration` to `AnimatorSet` we would either have to constraint the AnimatorSet to max duration or use the fallback as is being done in this PR. 

However, it seems like this `constraintDurationTo` is not valid for this `fly` animation as we are explicitly calling `setDuration` to the `AnimatorSet`. We better add a new extension function
```
internal fun AnimatorSet.constraintTo(maxDuration: Long): AnimatorSet {
    if (duration > maxDuration) {
        val factor = maxDuration / duration
        duration *= factor
    }
    return this
}
```

@LukasPaczos

Reply:
In this case, why don't we clamp the duration we set to the animator set in the first place? For example:
```diff
diff --git a/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt b/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt
index a759e824a1..f668713139 100644
--- a/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt
+++ b/libnavui-maps/src/main/java/com/mapbox/navigation/ui/maps/camera/transition/MapboxNavigationCameraTransition.kt
@@ -18,6 +18,7 @@ import com.mapbox.navigation.ui.maps.camera.utils.projectedDistance
 import com.mapbox.navigation.ui.maps.camera.utils.screenDistanceFromMapCenterToTarget
 import com.mapbox.navigation.utils.internal.ifNonNull
 import kotlin.math.abs
+import kotlin.math.min
 
 private const val LINEAR_ANIMATION_DURATION = 1000L
 private const val MAXIMUM_LOW_TO_HIGH_DURATION = 3000L
@@ -314,8 +315,8 @@ class MapboxNavigationCameraTransition(
             val zoomDelta = abs(currentZL - targetZL) * 80
             duration = normalizeProjection(projection + zoomDelta).toLong()
         }
+        duration = min(duration, transitionOptions.maxDuration)
         return createAnimatorSetWith(animators)
             .setDuration(duration)
-            .constraintDurationTo(transitionOptions.maxDuration)
     }
 }
```

Reply:
Yeah we could do that as well. 👍🏻

Reply:
Is there a reason to not clamp this inside `constraintDurationTo`?

The name of that function, seems like it should keep the duration within the bounds

Reply:
constraintDurationTo constraints the duration for child animators and not the parent animator set. Will leave the solution as is.